src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -211,6 +216,39 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
       );
   }
 
+  passwordToken(email: string, password: string): Observable<NbAuthResult> {
+    const url = this.getActionEndpoint('password');
+
+    return this.http.post(url, this.buildPasswordRequestData(email, password))
+      .pipe(
+        map((res) => {
+          return new NbAuthResult(
+            true,
+            res,
+            this.getOption('redirect.success'),
+            [],
+            this.getOption('defaultMessages'),
+            this.createToken(res));
+        }),
+        catchError((res) => {
+          const errors = [];
+          if (res instanceof HttpErrorResponse) {
```

Comment:
let's make it like this
```
if (res instanceof HttpErrorResponse) {
            errors = this.getOption('defaultErrors');
          } else {
            errors.push('Something went wrong.');
          }
```
in this case `Something went wrong` message indicates that this is not an HTTP error, so most likely an internal error happened on a client side.

Reply:
OK for pushing "Something went wrong" in case of error being internal and not HTTP.
But i would like to return the error message contained in the 401 http response, as described in RFC.

So i propose something like this:
```
if (res instanceof HttpErrorResponse) {
            if (res.error.error_description) {
              errors.push(res.error.error_description);
            } else {
              errors = this.getOption('defaultErrors');
            }
          } else {
            errors.push('Something went wrong.');
          }
```

Reply:
Yes, I agree, that perfectly makes sense. The only thing is that we probably need to modify the other method to conform with this style (so that all other requests will firstly look to errors into the responses, then in the config and only then report `something went wrong`.
So that it makes sense to move the error extraction into the additional function, doesn't it?

Reply:
gonna push lines below
```
  protected handleResponseError(res: any): Observable<NbAuthResult> {
    let errors = [];
    if (res instanceof HttpErrorResponse) {
      if (res.error.error_description) {
        errors.push(res.error.error_description);
      } else {
        errors = this.getOption('defaultErrors');
      }
    } else {
      errors.push('Something went wrong.');
    }
    return observableOf(
      new NbAuthResult(
        false,
        res,
        this.getOption('redirect.failure'),
        errors,
        [],
      ));
  }
```